src/code/TextTableParser.cs

```diff
@@ -0,0 +1,453 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Management.Automation;
+using System.Text.Json;
+
+namespace TextTableParser
+{
+    public class ColumnInfo {
+        public int Start;
+        public int Length;
+        public int SpaceLength;
+    }
+
+    public class HeaderField {
+        public string Name;
+        public int Start;
+        public int Length;
+    }
+
+    public class ObjectPropertyInfo {
+        public HeaderField[] HeaderFields;
+        ObjectPropertyInfo(string line, ColumnInfo[] cInfos)
+        {
+            HeaderFields = new HeaderField[cInfos.Length];
+            for(int i = 0; i < cInfos.Length; i++)
+            {
+                HeaderFields[i] = new HeaderField(){
+                    Name = line.Substring(cInfos[i].Start, cInfos[i].Length).Trim(),
+                    Start = cInfos[i].Start,
+                    Length = cInfos[i].Length
+                };
+
+            }
+        }
+    }
+
+    [Cmdlet("Convert","TextTable")]
+    public class ConvertTextTableCommand : PSCmdlet
+    {
+        [Parameter()]
+        public int MaximumWidth { get; set; } = 200;
+
+        [Parameter()]
+        public int AnalyzeRowCount { get; set; } = 0; // 0 is all
+
+        [Parameter(ValueFromPipeline=true,Mandatory=true,Position=0)]
+        [AllowEmptyString()]
+        public string[] Line { get; set; }
+
+        [Parameter()]
+        public int Skip { get; set; } = 0;
+
+        [Parameter()]
+        public SwitchParameter NoHeader { get; set; }
+
+        [Parameter()]
+        public SwitchParameter AsJson { get; set; }
+
+        [Parameter()]
+        public int[] ColumnOffset { get; set; }
+
+        [Parameter()]
+        public SwitchParameter Convert { get; set; }

```

Comment:
I don't have a good suggestion on this but the parameter name is ambiguous especially since the verb is `Convert` maybe something like `ConvertPropertyValue` or `ConvertPropertyType` or `CastPropertyValue` or `CastPropertyType`.

Something to indicate from the parameter name what is being changed in the output.